### DISCUSSION TITLE
Real-Time Updater before-after message

### DISCUSSION THREAD

**Discussion by SalvadorCali** [mia-platform/community#138] [Q&A] [answered]
Hi, everybody!
Under [this paragraph](https://docs.mia-platform.eu/docs/fast_data/configuration/realtime_updater/common#message-schema) we found this disclaimer inside the danger box:

> For performance reasons, the projection state before of the projection is not returned after an update operation, because a costly find operation would be necessary to retrieve such information from the database. The only exception is an UPDATE resulting from a virtual delete: in this case the before is returned entirely.

After an **update** or a **physical delete** is the field `before` left empty or only the property `__STATE__` of the projection is omitted?

**Comment by danibix95** [comment, marked as answer]
Hi @SalvadorCali,

I verified the code and the `before` property is not set (either it gets an `undefined` or `null` value) when an _insert_, _update_ or _physical delete_ operation is performed.

When the documentation mentions

> the projection state before of the projection

it means the value of the projection before the _insert_, _update_ or _physical delete_ operation occurs.  
Usually, across the documentation, when it is intended the property `__STATE__` of a projection record, it is explicitly written with the leading and trailing underscores.
